src/main/java/org/opensearch/neuralsearch/query/NeuralSparseQueryBuilder.java

```diff
@@ -376,4 +417,58 @@ public String getWriteableName() {
     private static boolean isClusterOnOrAfterMinReqVersionForDefaultModelIdSupport() {
         return NeuralSearchClusterUtil.instance().getClusterMinVersion().onOrAfter(MINIMAL_SUPPORTED_VERSION_DEFAULT_MODEL_ID);
     }
+
+    private Map<String, Float> getAllTokens() {
+        Map<String, Float> queryTokens = queryTokensSupplier.get();
+        validateQueryTokens(queryTokens);
+        return queryTokens;
+    }
+
+    private Map<String, Float> getHighScoreTokens(Map<String, Float> queryTokens, float ratio) {
+        return getFilteredScoreTokens(queryTokens, true, ratio);
+    }
+
+    private Map<String, Float> getLowScoreTokens(Map<String, Float> queryTokens, float ratio) {
+        return getFilteredScoreTokens(queryTokens, false, ratio);
+    }
+
+    private Map<String, Float> getFilteredScoreTokens(Map<String, Float> queryTokens, boolean aboveThreshold, float ratio) {
+        float max = queryTokens.values().stream().max(Float::compare).orElse(0f);
+        float threshold = ratio * max;
+        if (max == 0) {
+            return Collections.emptyMap();
+        }
+        return queryTokens.entrySet()
+            .stream()
+            .filter(entry -> (aboveThreshold == (entry.getValue() >= threshold)))
+            .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+        // This call will generate a filter from queryTokens.
+        // When aboveThreshold is true, will filter out all key-value pairs whose value >= threshold to a return map.
+        // When aboveThreshold is false, will filter out all key-value pairs whose value <= threshold to a return map.
+    }
+
+    private BooleanQuery buildFeatureFieldQueryFromTokens(Map<String, Float> tokens, String fieldName) {
+        BooleanQuery.Builder builder = new BooleanQuery.Builder();
+        for (Map.Entry<String, Float> entry : tokens.entrySet()) {
+            builder.add(FeatureField.newLinearQuery(fieldName, entry.getKey(), entry.getValue()), BooleanClause.Occur.SHOULD);
+        }
+        return builder.build();
+    }
+
+    private void completeTwoPhasePara(IndexSettings indexSettings) {
```

Comment:
Please fix formatting in this method, every method call should be inside the curly braces and on a new line